core/base/src/TROOT.cxx

```diff
@@ -2737,6 +2737,31 @@ void TROOT::SetMacroPath(const char *newpath)
       macroPath = newpath;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// If `web` is not empty it defines where web graphics should be rendered
+///
+/// - "cef" web graphics is rendered via the Chromium Embedded Framework
+/// - "qt5" web graphics is rendered via Qt5
+/// - "browser" or "" web graphics is rendered in the default web browser
+///
+/// It also turn the normal graphics to Batch to avoid the loading of local
+/// graphics libraries.
+
+void TROOT::SetWeb(TString web)
+{
+   fWeb = web;
+   fWeb.ToLower();
+   if (fWeb.Contains("cef") || fWeb.Contains("qt5") || fWeb.Contains("browser") || fWeb.Length() ==0) {
```

Comment:
Where are much more variants as "cef" or "qt5" or "browser".
One should have another arguments list in the method. 
For instance, ``const char*`` or ``const std::string &``:
   * empty - default web display (depends from compiled components)
   * <progname> - program name for display, one could have some expansion in .rootrc file

One also need extra argument, which specifies batch mode of web display.

Reply:
Why is "batch mode of web display" a specific setting?
Wouldn't:
```
gROOT->IsBatch() && gROOT->IsWebDisplay()
```
be sufficient?

Reply:
> gROOT->IsBatch() && gROOT->IsWebDisplay()

It is not enough. When running web display, we want fully suppress normal graphics.
To avoid changes in many different places, we just set gROOT->SetBatch(kTRUE) for that.

But webdisplay needs to know, if  batch or not-batch mode should be used.

Reply:
This is/was fine for a prototyping phase but at this point I think we need to do better.  As is this plan will be very confusing to user.  For example they will test 'gROOT->IsBatch()' and that will say 'true' but would still display stuff in a browser ...

Reply:
We are still in prototyping.
This workaround should allow us to test different web modes (cef/qt5/browsers) on all platforms.
Already now there is combination (Mac/qt5), which works only when graphics not initialized. 
Once all variants are tested, one could propose final implementation.

Yes, in such final implementation IsBatch() should be used also for web display modes. But then in many different places we must ensure, that Cocoa/x11/win not initialized if IsWebDisplay() returns true.